sentry/src/main/java/io/sentry/TransactionContexts.java

```diff
@@ -0,0 +1,41 @@
+package io.sentry;
+
+import io.sentry.protocol.Contexts;
+import io.sentry.protocol.SentryId;
+import org.jetbrains.annotations.NotNull;
+
+public final class TransactionContexts extends Contexts {
+  private static final long serialVersionUID = 252445813254943011L;
+
+  public TransactionContexts() {
+    this(new Trace());
+  }
+
+  private TransactionContexts(final @NotNull Trace trace) {
+    this.setTrace(trace);
+  }
+
+  /**
+   * Creates {@link TransactionContexts} from traceparent header.
+   *
+   * @param traceparent - the traceparent header
+   * @return the transaction contexts
+   */
+  public static @NotNull TransactionContexts fromTraceparent(final @NotNull String traceparent) {
+    final String[] parts = traceparent.split("-", -1);
+    if (parts.length < 2) {
```

Comment:
Throwing from a public API.. 

We have to make sure we capture this and send an event and don't break user code

Maybe we need checked exceptions here?

Reply:
Agreed!